pkg/kv/kvserver/replica_application_result.go

```diff
@@ -510,8 +510,7 @@ func (r *Replica) handleTruncatedStateResultRaftMuLocked(
 	// v22.1 that added it, when all truncations were strongly coupled. It is not
 	// safe to consider the log size delta trusted in this case. Conveniently,
 	// this doesn't need any special casing.
-	expectedFirstIndexWasAccurate := r.shMu.raftTruncState.Index+1 == pt.expectedFirstIndex
-	isRaftLogTruncationDeltaTrusted := pt.isDeltaTrusted && expectedFirstIndexWasAccurate
+	isDeltaTrusted := pt.isDeltaTrusted && r.shMu.raftTruncState.Index+1 == pt.expectedFirstIndex
```

Comment:
here it is _almost_ better to instead mutate `pt.isDeltaTrusted` so that there aren't two identifiers that mean different things but have the right name. I'm not saying it is better. Just thinking out loud.